src/snowflake/snowpark/dataframe.py

```diff
@@ -2060,6 +2060,21 @@ def join(
                 using_columns = [using_columns]
             elif isinstance(using_columns, Column):
                 using_columns = using_columns
+            elif (
+                isinstance(using_columns, list)
```

Comment:
The type hints says `Iterable` instead of `list`. Better support `Iterable` so list, tuple and other iterables are supported.